buildpacks/builder.go

```diff
@@ -119,6 +119,11 @@ func (builder *Builder) Build(ctx context.Context, f fn.Function) (err error) {
 		}
 	}
 
+	buildEnvs := make(map[string]string, len(f.BuildEnvs))
+	for _, env := range f.BuildEnvs {
+		buildEnvs[*env.Name] = *env.Value
+	}
+
 	packOpts := pack.BuildOptions{
 		AppPath:        f.Root,
 		Image:          f.Image,
```

Comment:
Let me check, I actually rebased it.

Reply:
Yep, must be what happened. I also can't see the history since I rewrote it. Let me test out my changes once just to be sure everything's working fine

Reply:
```diff
                Builder:        packBuilder,
                Buildpacks:     f.Buildpacks,
+               Env:                buildEnvs,
                TrustBuilder: !daemonIsPodmanBeforeV330 &&
                        (strings.HasPrefix(packBuilder, "quay
```

Reply:
Have another bug here. Missed processing the env to handle values that are set in environment where `func build` is run. I'm mimicking the processing done in `docker/runner.go`

Reply:
Fixed it and verified buildEnvs work in both cases 
- hardcoded in func.yaml
- set as a local environment variable where func build is run